spring-integration-jdbc/src/main/java/org/springframework/integration/jdbc/store/JdbcChannelMessageStore.java

```diff
@@ -409,7 +422,8 @@ public void afterPropertiesSet() throws Exception {
 	@Override
 	public MessageGroup addMessageToGroup(Object groupId, final Message<?> message) {
 		try {
-			this.jdbcTemplate.update(getQuery(this.channelMessageStoreQueryProvider.getCreateMessageQuery()),
+			this.jdbcTemplate.update(getQuery(Query.CREATE_MESSAGE,
+						() -> this.channelMessageStoreQueryProvider.getCreateMessageQuery()),
```

Comment:
What is just wrong with calling the since it just returns a constant anyway?

Reply:
Because there is no need to call every time - the supplier is only invoked if the cache does not already contain the key. Even though it only returns a constant, it's an unnecessary method call.